validator/rules/no-native-types.js

```diff
@@ -20,25 +20,45 @@ import { ESLintUtils } from '@typescript-eslint/utils';
 
 const createRule = ESLintUtils.RuleCreator(name => `https://example.com/rule/${name}`)
 
-const TYPES_TO_AVOID = ['Record', 'Partial', 'Required', 'Pick', 'Omit'];
+const UTILITY_TYPES = ['Record', 'Partial', 'Required', 'Pick', 'Omit'];
+
+const COLLECTION_TYPES = ['Map', 'Set', 'WeakMap', 'WeakSet'];
+
+const TYPES_TO_AVOID = [...UTILITY_TYPES, ...COLLECTION_TYPES];
+
+const TYPE_SUGGESTIONS = {
+  'Record': 'Use Dictionary instead',
+  'Map': 'Use Dictionary instead',
+  'Set': 'Use an array type instead (e.g., string[])',
+  'WeakMap': 'Use Dictionary instead',
+  'WeakSet': 'Use an array type instead',
+};
```

Comment:
When trying to add `Array` to the list, I added it to `TYPE_SUGGESTIONS` and realized it had no effect. Can we please stop using `TYPES_TO_AVOID` and `TYPE_SUGGESTIONS` and only rely on `TYPE_SUGGESTIONS`? Perhaps that will enable us to provide more helpful suggestions than "Use spec-defined aliases instead". (There's no equivalent to `Omit` or `Pick`, as far as I know.)